### PR TITLE
fix: add disabled prop to x icon

### PR DESCRIPTION
**Pull request by Sandtory**
Add disabled prop to x icon so that it will not let user clear the search field when used

### CODE REVIEW

File: packages/components/src/components/Search/Search.tsx [hunk 1]
```diff
@@ -153,6 +153,7 @@ export const Search = ({
                                     color="primary"
                                     variant="ghost"
                                     onPress={handleClearText}
+                                    disabled={disabled}
```
Comment (HaakonSvane) [inline packages/components/src/components/Search/Search.tsx:156]: Should we maybe instead just don't show this clear button if the search bar is disabled? 
Could potentially just add the disabled check next to the condition for rendering this on line 148 in that case. Thoughts?

**Review by HaakonSvane — APPROVED** [review]
Nice 😠 
Check my comment regarding the rendering of the button and change if you think it is good idea.

File: packages/components/src/components/Search/Search.tsx (hunk 1 above)
Comment (Sandtory) [reply packages/components/src/components/Search/Search.tsx:156]: yes that is smoother and more highly advanced senior code, i agree wholeheartedly 😡

File: packages/components/src/components/Search/Search.tsx [hunk 2]
```diff
@@ -145,7 +145,7 @@ export const Search = ({
                         </View>
                     }
                     rightAdornments={
-                        text ? (
```
Comment (HaakonSvane) [inline packages/components/src/components/Search/Search.tsx:148]: Isn't it the other way around? 
`text && !disabled` ?
Now it seems to only render the clear button if the search contains text AND is disabled

**Review by HaakonSvane — CHANGES_REQUESTED** [review]

**Review by HaakonSvane — APPROVED** [review]
😠 perfect

**Pull request merged**